tests/libraries/dynamic_ref/test.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+set -o pipefail
+
+rlJournalStart
+    rlPhaseStartSetup "Prepare git repos"
+        testdir=$PWD
+        rlRun "repo1=\$(mktemp -d)"
+        rlRun "repo2=\$(mktemp -d)"
+        # create test repo
+        rlRun "cp data/test.sh $repo1"
+        cat <<EOF > $repo1/main.fmf
+/first:
+  test: bash test.sh
+  framework: beakerlib
+  require:
+  - name: /dynref
+    type: library
+    url: $repo2
+    ref: "@dynamic-ref"
+EOF
+        rlRun "tmt init $repo1"
+        rlRun "pushd $repo1"
+        rlRun "git init"
+        rlRun "git config --local user.email me@localhost.localdomain"
+        rlRun "git config --local user.name m e"
+        rlRun "git add -A"
+        rlRun "git commit -m initial"
+        rlRun "popd"
+        # create library repo
+        rlRun "tmt init $repo2"
+        rlRun "pushd $repo2"
+        rlRun "git init"
+        rlRun "git config --local user.email me@localhost.localdomain"
+        rlRun "git config --local user.name m e"
+        rlRun "cp $testdir/data/dynamic-ref $repo2"
+        rlRun "git add -A"
+        rlRun "git commit -m first"
+        rlRun "git checkout -b otherbranch"
+        rlRun "mkdir dynref"
+        rlRun "cp -r $testdir/data/lib.sh $repo2/dynref"
+        cat <<EOF > $repo2/dynref/main.fmf
+test: bash lib.sh
+framework: beakerlib
+EOF
+        rlRun "git add -A"
+        rlRun "git commit -m second"
+        rlRun "git checkout master"
+        rlRun "popd"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Search for lib in a dynamically resolved branch"
+        rlRun "pushd $repo1"
+        rlRun -s "tmt --context foo=1 run --rm -a -vvv -ddd discover provision -h local"
+        rlAssertGrep "Dynamic 'ref' definition file '.*' detected." $rlRun_LOG -E
+        rlAssertGrep "Dynamic 'ref' resolved as 'otherbranch'." $rlRun_LOG
+        rlAssertGrep "total: 1 test passed" $rlRun_LOG
+        rlRun "popd"
+    rlPhaseEnd
+
+    rlPhaseStartCleanup
+        rlRun "rm -rf $repo1" 0 "Remove temporary directories"
```

Comment:
```suggestion
        rlRun "rm -rf $repo1 $repo2" 0 "Remove temporary directories"
```

Reply:
Yes

Reply:
Fixed in https://github.com/teemtee/tmt/commit/c376cf1ccfffe59feb4396c0c82a36bd90762cf4.